pkg/gits/helpers.go

```diff
@@ -499,14 +499,24 @@ func ForkAndPullRepo(gitURL string, dir string, baseRef string, branchName strin
 		}
 	}
 
-	err = gitter.FetchBranch(dir, originRemote, branchName)
+	userDetails := provider.UserAuth()
+	originFetchURL, err := gitter.CreateAuthenticatedURL(originURL, &userDetails)
+	if err != nil {
+		return "", "", nil, nil, errors.Wrapf(err, "failed to create authenticated fetch URL for %s", originURL)
+	}
+	err = gitter.FetchBranch(dir, originFetchURL, branchName)
```

Comment:
@pmuir so this code does something similar as the push code we have been talking about. It effectively uses the fully qualified remote URL + credentials for the fetch. This works fine over HTTP, but fails for file:// URLs. Hence all the test failures. Not quite sure how to move forward with this. 

Any suggestions? I cannot figure out how to adjust the tests, adding another if statement to detect whether we deal with http vs file seems fugly and when I tried did not resolve all test failures. 

After all the credential helper might be after all the way forward. WDYT?

Reply:
I think I found a way, let's see whether tests are passing now